src/ImageKit/ImageKit.php

```diff
@@ -437,42 +661,75 @@ public function bulkFileDeleteByIds($options)
+    public function bulkDeleteFiles($fileIds=null)
     {
+        if (!isset($fileIds)) {
+            return Response::respond(true, ((object)ErrorMessages::$fileIdS_MISSING));
+        }
+        if (!is_array($fileIds)) {
+            return Response::respond(true, ((object)ErrorMessages::$fileIdS_NON_ARRAY));
+        }
+        if (sizeof($fileIds)==0) {
+            return Response::respond(true, ((object)ErrorMessages::$fileIdS_EMPTY_ARRAY));
+        }
+        
         $this->httpClient->setUri(Endpoints::getDeleteByFileIdsEndpoint());
         return Manage\File::bulkDeleteByFileIds($fileIds, $this->httpClient);
     }
 
     /**
-     * This will copy a file from one location to another. This method accepts the source file's path and destination
+     * This will copy a file from one location to another. This method accepts an array of source file's path, destination path and boolean to include versions or not
      * folder path.
      *
      * @link https://docs.imagekit.io/api-reference/media-api/copy-file
      *
-     * @param $sourceFilePath
-     * @param $destinationPath
+     * @param $parameter['sourceFilePath','destinationPath','includeVersions']
      * @return Response
      */
-    public function copyFile($sourceFilePath, $destinationPath)
+    public function copyFile($parameter=null)
     {
+        if(!isset($parameter)){
+            return Response::respond(true, ((object)ErrorMessages::$COPY_FILE_PARAMETER_MISSING));
+        }
+        if(!is_array($parameter)){
+            return Response::respond(true, ((object)ErrorMessages::$COPY_FILE_PARAMETER_NON_ARRAY));
+        }
+        if(sizeof($parameter)==0){
+            return Response::respond(true, ((object)ErrorMessages::$COPY_FILE_PARAMETER_EMPTY_ARRAY));
+        }
+        if (empty($parameter['sourceFilePath']) || empty($parameter['destinationPath']) || !isset($parameter['includeVersions'])) {
+            return Response::respond(true, ((object)ErrorMessages::$COPY_FILE_DATA_INVALID));
+        }
+
         $this->httpClient->setUri(Endpoints::getCopyFileEndpoint());
-        return Manage\File::copy($sourceFilePath, $destinationPath, $this->httpClient);
+        return Manage\File::copy($parameter['sourceFilePath'], $parameter['destinationPath'], $parameter['includeVersions'], $this->httpClient);
     }
 
     /**
-     * This will move a file from one location to another. This method accepts the source file's path and destination
+     * This will move a file from one location to another. This method accepts an array containing source file's path and destination path
      * folder path.
      *
      * @link https://docs.imagekit.io/api-reference/media-api/move-file
      *
-     * @param $sourceFilePath
-     * @param $destinationPath
+     * @param $parameter['sourceFilePath','destinationPath']
      * @return Response
      */
-    public function moveFile($sourceFilePath, $destinationPath)
+    public function moveFile($parameter=null)
```

Comment:
Should we preserve the old method as well to not introduce breaking changes? However, in docs, we will only write this new convention. Internally both will continue to work.
The same comment is applicable to all such breaking changes. Generally, we should avoid breaking changes unless they are absolutely necessary for sanity or consumer benefit.

Reply:
preserved the following old methods:
`copyFile`, `moveFile`, `renameFile`

but cannot preserve the folder related methods:
`copyFolder`, `moveFolder`
becuase the number of parameters are different and cannot create other method becuase of method name mismatch.

Reply:
The parameter in question `includeVersions` is new and was never mentioned with the old method. So if the old (and undocumented) method does not support it, it should be fine.